db/migrate/20190128014059_create_users.rb

```diff
@@ -1,11 +1,16 @@
 class CreateUsers < ActiveRecord::Migration[5.2]
   def change
-    create_table :users do |t|
-      t.string :display_name
-      t.string :uid
-      t.string :email
+    create_table :users, id: :uuid do |t|
+      t.string :display_name, null: false
+      t.string :firebase_id, null: false
+      t.string :email, null: false
+      t.boolean :email_verified, default: false
+      t.datetime :last_seen_at
```

Comment:
When is this updated? Also is this necessary?

Reply:
Check out the `Authentication` concern below.

It's useful to know which users are inactive - e.g. to purge inactive user accounts, or to re-engage.

Reply:
Shoot I just glanced right past it!

Reply:
That was the main file in this PR! 😄